### DISCUSSION TITLE
Question about symmetric boundary condition setting of thermal stress calculation of a 1/8 rotational symmetric model

### DISCUSSION THREAD

**Discussion by WayneX-Wong** [idaholab/moose#27498] [Q&A General]
Hello MOOSE experts:

I simulated a 1/8 rotational symmetric solid mechanical problem, but ran into following question:

**Precondition:**
I've simulated a thermal conduction problem using this symmetric model, and got temperature distribution, which I was about to use for thermal stress calculation

**My setting:**
I set zero normal-displacement on  both symmetric surfaces like:
$$\vec{n}\bullet\vec{disp}=0\ \ \Rightarrow\ \ n_x{disp}_x+n_y{disp}_y=0$$
And I used `MatchValueBC` for realizing this:
![input](https://github.com/idaholab/moose/assets/111510550/5aafcc9f-cea5-4af0-8c57-bf922dfefa9a)

**Result:**
There seems still normal displacement on symmetric surface:
![model](https://github.com/idaholab/moose/assets/111510550/89cc3e06-e5a7-4ba3-bf0c-7d41a84058bc)
Where white block is original model, and wireframe is displaced geometry, two green lines stands for original symmetric surface.
And I've plotted displacement of symmetry surface:
![plot](https://github.com/idaholab/moose/assets/111510550/b951dea5-b3c2-415e-8bbc-590573b5e120)
Where `Result` is 
$$n_x{disp}_x+n_y{disp}_y$$
and doesn't equal to 0.

**Question**
It seems my symmetric boundary condition didn't work. What might be the reason for this?
Or if there's other way to set this kind of symmetric boundary condition?

Thanks! (Sorry I can only provide relative photos rather than relative input files, cause I ran this example on a LAN computer)

**Comment by GiudGiud** [comment]
Hello

If you are not using parallel processing you can use the Functor system (the functor APIs on a variable) to tie the values on the top surface to the values on the bottom one. 
You can use an ElemPointArg and specify the point analytically to go from one surface to the other

You can keep trying on constrainting the displacement on a single surface by writing a penalty version of this BC

Guillaume

**Reply by WayneX-Wong** [reply]
Hello Guillaume, thanks for reply, while it's hard for me to fully understand your suggestion, I might need to learn Functor system better.

But I still wonder what's wrong with my settings by using `MatchValueBC`, could you help me point out the problem?

**Reply by GiudGiud** [reply]
I dont know.
Did you double check those two coefficients?
Do you get a converged result on every time step? What does the simulation log look like?

**Reply by WayneX-Wong** [reply]
I check those coefficients, and the calculation did converge with relative tolerance below than1e-10.

Besides I refine the mesh but same strange result.

**Reply by GiudGiud** [reply]
are rel_fuel1 / 2 both exactly on the outer surface? sometimes the mesher will place nodes inside the volume in a nodeset

have you tried using approaches from the solid mechanics module? I m thinking they have more custom techniques for this
https://mooseframework.inl.gov/source/bcs/PenaltyInclinedNoDisplacementBC.html

**Reply by WayneX-Wong** [reply]
I will try to find more relevant from Solid Mechanics module.

Thanks again for your suggestions!
